include/solo5/solo5.h

```diff
@@ -245,37 +254,41 @@ struct solo5_block_info {
 };
 
 /*
- * Retrieves information about the block device. Caller must supply space for
- * struct solo5_block_info in (info).
+ * Acquires a handle to the block device declared as (name) in the
+ * application manifest. The returned handle is stored in (*handle), and
+ * properties of the block device are stored in (*info). Caller must supply
+ * space for struct solo5_block_info in (info).
  */
-void solo5_block_info(struct solo5_block_info *info);
+solo5_result_t solo5_block_acquire(const char *name, solo5_handle_t *handle,
+        struct solo5_block_info *info);
 
 /*
- * Writes data of (size) bytes from the buffer (*buf) to the block device,
- * starting at byte (offset). Data is either written in it's entirety or not at
- * all ("short writes" are not possible).
+ * Writes data of (size) bytes from the buffer (*buf) to the block device
+ * identified by (handle), starting at byte (offset). Data is either written in
+ * it's entirety or not at all ("short writes" are not possible).
```

Comment:
Is this really a physical guarantee we can make? Isn't it more like "short writes will result in an error" ? (ie when the device connection is flaky, like if it's on NFS, or the flash is a bit faulty)